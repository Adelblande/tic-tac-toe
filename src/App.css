#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

div.board {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 1rem;
}

div.board span {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100px;
  height: 100px;
  font-size: 2rem;
  background-color: #333;
  color: #fff;
  cursor: pointer;
}
