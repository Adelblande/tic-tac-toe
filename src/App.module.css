#root {
  width: 100%;
  height: 100vh;
  margin: 0 auto;
}

.message {
  font-size: 1.5rem;
  color: #fff;
  text-align: center;
  margin: 16px 0;
}
