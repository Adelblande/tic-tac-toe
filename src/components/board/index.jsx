import styles from "./styles.module.css";
import { useSettings } from "../../hooks/use-settings";

export function Board({ game, handleClick }) {
  const { colors } = useSettings();

  return (
    <div className={styles.board}>
      {game.map((value, index) => {
        return (
          <span
            id={`cell-${index}`}
            key={index}
            style={{ color: colors[value] }}
            onClick={() => handleClick(index)}
          >
            {value}
          </span>
        );
      })}
    </div>
  );
}
