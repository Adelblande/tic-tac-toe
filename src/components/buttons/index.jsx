import styles from "./styles.module.css";

export function ResetButton({ value, onClick }) {
  return (
    <button className={styles.resetButton} onClick={onClick}>
      {value}
    </button>
  );
}
