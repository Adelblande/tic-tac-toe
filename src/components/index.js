export * from "./board";
export * from "./buttons";
export * from "./scores";
export * from "./timer";
