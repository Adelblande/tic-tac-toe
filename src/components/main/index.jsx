import styles from "./styles.module.css";

export function Main({ children, backgroundColor }) {
  return (
    <main className={styles.main} style={{ backgroundColor }}>
      {children}
    </main>
  );
}
