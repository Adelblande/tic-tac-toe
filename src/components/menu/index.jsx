import styles from "./styles.module.css";
import { useSettings } from "../../hooks/use-settings";

export function Menu() {
  const { availableColorsCombinations, setColors } = useSettings();

  function toggle() {
    if (
      document.querySelector(`.${styles.set}`).classList.contains(styles.show)
    ) {
      document.querySelector(`.${styles.set}`).classList.remove(styles.show);
    } else {
      document.querySelector(`.${styles.set}`).classList.add(styles.show);
    }
  }

  return (
    <div className={styles.set}>
      <button className={styles.mainMenu} onClick={toggle}>
        <img src="/settings.png" alt="Menu settings" width={30} height={30} />
      </button>
      <ul>
        {availableColorsCombinations.map((colors, index) => (
          <li key={index}>
            <button onClick={() => setColors(colors)}>
              <span style={{ color: colors.x }}>x</span>
              <span style={{ color: colors.o }}>o</span>
              <span style={{ color: colors.board }}>#</span>
            </button>
          </li>
        ))}
      </ul>
    </div>
  );
}
