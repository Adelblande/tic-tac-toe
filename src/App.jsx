import "./App.css";
import { ResetButton, Timer, Board, Scores } from "./components/index.js";
import { useGame } from "./hooks/use-game";

function App() {
  const { game, message, player, scores, handleClick, resetGame, setPlayer } =
    useGame();

  return (
    <>
      <header className="header">
        <Scores scores={scores} />
        {!message && <Timer player={player} setPlayer={setPlayer} />}
      </header>

      <main className="main">
        {message && <span className="message">{message}</span>}
        <Board game={game} handleClick={handleClick} />
        <ResetButton onClick={resetGame} value="Reset Game" />
      </main>
    </>
  );
}

export default App;
