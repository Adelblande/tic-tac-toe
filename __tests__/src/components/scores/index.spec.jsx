import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { Scores } from "../../../../src/components";

describe("Scores", () => {
  it("should render scores components", () => {
    const { container } = render(<Scores scores={{ x: 0, o: 0 }} />);

    expect(container).toMatchSnapshot();
  });
});
