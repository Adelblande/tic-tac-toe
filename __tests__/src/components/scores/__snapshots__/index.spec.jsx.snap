// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Scores > should render scores components 1`] = `
<div>
  <div
    class="_scores_afa92b"
  >
    <h3>
      Scores
    </h3>
    <p>
      Player X: 
      0
    </p>
    <p>
      Player O: 
      0
    </p>
  </div>
</div>
`;
